app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,142 @@
+package org.oppia.app.player.state;
+
+import android.text.Spannable
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.databinding.library.baseAdapters.BR
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.item_selection_interaction_items.view.*
+import kotlinx.android.synthetic.main.multiple_choice_interaction_items.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.ItemSelectionInteractionItemsBinding
+import org.oppia.app.databinding.MultipleChoiceInteractionItemsBinding
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.player.state.listener.InteractionAnswerRetriever
+import org.oppia.app.player.state.listener.InteractionListener
+import org.oppia.util.parser.HtmlParser
+
+const val VIEW_TYPE_MULTIPLE_CHOICE = 1
+const val VIEW_TYPE_ITEM_SELECTION = 2
+
+/** Adapter to bind the interactions to the [RecyclerView]. It handles MultipleChoiceInput and ItemSelectionInput interaction views. */
+class InteractionAdapter(
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
+  private val explorationId: String,
+  val itemList: Array<String>?,
+  private val interactionId: String
+  ) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), InteractionAnswerRetriever {
+
+  private var mSelectedItem = -1
```

Comment:
Change this to `selectedItem` prefer not using `m`

Reply:
done.